src/components/document/TableView/TableView.jsx

```diff
@@ -1,6 +1,78 @@
-const TableView = () => { 
-    return <div>
-        TableView
+import CheckIcon from "@/assets/check_fill.svg";
+import CloseIcon from "@/assets/close_fill.svg";
+import CommentIcon from "@/assets/comment_fill.svg";
+import NavigateToIcon from "@/assets/navigate_to_icon.svg";
+import { Slider } from 'antd';
+import { useState } from "react";
+
+const RedactionFilterDropdownMenu = () => {
+    //TODO: Some sort of dropdown
+    return <div className="w-full text-end">
+        Accepted Suggestions 
+    </div>
+}
+
+const splitText = (document) => {
+    return document.body.children.map((paragraph) => {
+        return paragraph.children
+    }).flat(1)
+}
+
+const TableEntry = (sentence) => {
+    const text = sentence.sentence.text
+    // Temporarily insert " REDACTED " into the sentence
+    const sentenceWithRedactedWord = text.slice(0, text.length/2) + " REDACTED " + text.slice(text.length/2)
+    const words = sentenceWithRedactedWord.split(' ')
+    const redactedWordIndex = words.indexOf("REDACTED")
+    const max = Math.max(words.length - redactedWordIndex - 1, redactedWordIndex)
+    const min = 0
+    const [sentenceLength, setSentenceLength] = useState(0)
+
+    const SentenceWithRedaction = () => {
+        //Separate sentence into preRedactedSection, redactedWord, and postRedactedSection
+        const preRedactedSection = words.slice(Math.max(redactedWordIndex-sentenceLength,0), redactedWordIndex).join(' ')
+        const redactedWord = ' ' + words[redactedWordIndex] + ' '
+        const postRedactedSection = words.slice(redactedWordIndex + 1, redactedWordIndex+sentenceLength + 1).join(' ')
+
+        return <div className="m-2 flex-1 justify-center items-center flex text-center">
+            {[preRedactedSection, redactedWord, postRedactedSection]}
+        </div>
+    }
+    
+    const SentenceVisibilityScale = (props) => {
+        return <Slider {...props} onChange={setSentenceLength} value={sentenceLength} className="w-20 m-2 justify-center items-center flex"/>
+    }
+    
+    const TableActionButtons = () => {
+        return <div className="flex flex-row p-1">
+            <img src={CheckIcon}/>
+            <img src={CloseIcon}/>
+            <img src={CommentIcon}/>
+            <img src={NavigateToIcon}/>
+        </div>
+    }
+
+    
+    return <div className="border border-solid border-slate-100 flex flex-row p-2">
+        <SentenceWithRedaction className="m-2"/>
+        <SentenceVisibilityScale min={min} max={max}/>
+        <TableActionButtons/>
+    </div>
+}
+
+const TableView = ({ document }) => { 
+    const sentences = splitText(document)
+    const tableEntries = sentences.map((sentence, idx) => {
+        return <div key={idx}>
```

Comment:
hehehe it's a bit of an antipattern to use to `idx` as the key for iterations, since it messes up sorting and reordering. I'm not sure what a better key is though, maybe hash the sentence contents?? Or the sentence's index in the paragraph (rather than the iteration).